cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -66,15 +70,22 @@
 import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.ExecutionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 /**
  * Wrapper around the dataproc client that adheres to our configuration settings.
  */
 public class DataprocClient implements AutoCloseable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DataprocClient.class);
   // something like 2018-04-16T12:09:03.943-07:00
   private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss.SSSX");
 
+  private static final Pattern GCP_NETWORK_FULL_PATH =
+    Pattern.compile("(https://www.googleapis.com/compute/v1/projects/)(.*)?(/global/networks/)(.*)?");
```

Comment:
is 'global' here a location? Is it correct to hardcode it?

Reply:
Since its only for comparing with the system network, I think its ok to have hardcode in it as we always create the system network with the `global` location. Also in future we may run the cdap services with customer supplied service accounts in which case there is no guarantee that the service account will have access on the system network. Ideally we should pass the system configs to the provisioner so we dont have to have logic of figuring out system network or system project in the Dataproc provisioner itself right?